tools/dependency/validate.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python3
+"""Validate the relationship between Envoy dependencies and core/extensions.
+
+This script verifies that bazel query of the build graph is consistent with
+the use_category metadata in bazel/repository_locations.bzl.
+"""
+
+import re
+import subprocess
+import sys
+
+from importlib.util import spec_from_loader, module_from_spec
+from importlib.machinery import SourceFileLoader
+
+# bazel/repository_locations.bzl must have a .bzl suffix for Starlark import, so
+# we are forced to do this workaround.
+_repository_locations_spec = spec_from_loader(
+    'repository_locations',
+    SourceFileLoader('repository_locations', 'bazel/repository_locations.bzl'))
+repository_locations = module_from_spec(_repository_locations_spec)
+_repository_locations_spec.loader.exec_module(repository_locations)
+
+# source/extensions/extensions_build_config.bzl must have a .bzl suffix for Starlark
+# import, so we are forced to do this workaround.
+_extensions_build_config_spec = spec_from_loader(
+    'extensions_build_config',
+    SourceFileLoader('extensions_build_config', 'source/extensions/extensions_build_config.bzl'))
+extensions_build_config = module_from_spec(_extensions_build_config_spec)
+_extensions_build_config_spec.loader.exec_module(extensions_build_config)
+
+BAZEL_QUERY_EXTERNAL_DEP_RE = re.compile('@(\w+)//')
```

Comment:
i would put constants first after imports

Reply:
I think this one is fine, the constants come after the file-level setup, the dynamic .bzl load bits are morally equivalent to an import statement.